specification/sections/02-providers.md

```diff
@@ -150,3 +150,70 @@ class MyProvider implements Provider {
 #### Requirement 2.3.3
 
 > In cases of abnormal execution, the `resolution details` structure's `error message` field **MAY** contain a string containing additional detail about the nature of the error.
+
+### 2.4 Initialization
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 2.4.1
+
+> The `provider` interface **MUST** define an `initialize` function which accepts the global `evaluation context` as a argument, which can be optionally implemented to perform initialization logic relevant to the provider.
+
+```
+class MyProvider implements Provider {
+  //...
+
+  void initialize(EvaluationContext initialContext) {
+    /*
+      A hypothetical initialization function: connect to a flag database, then pass a callback which fires PROVIDER_READY
+    */
+    this.connectToDatabase(() => this.emitEvent(ProviderEvents.READY));
+  }
+
+  //...
+}
+```
+
+### 2.5. Provider context reconciliation
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+Client-focused providers may need a mechanism to understand when their cache of evaluated flags must be invalidated or updated. An `on-context-set` handler can be defined which performs whatever operations are needed to reconcile the evaluated flags with the new context.
```

Comment:
```suggestion
Single-context focused providers may need a mechanism to understand when their cache of evaluated flags must be invalidated or updated. An `on-context-set` handler can be defined which performs whatever operations are needed to reconcile the evaluated flags with the new context.
```
Perhaps I've misunderstood but it seems to me that `single-context` encompasses client?

Reply:
Ya, I think I can change the language.

I'm also wondering if we should make this requirement conditional on single-context or not. Open to thoughts on that.